epubmaker_preprocessing.rb

```diff
@@ -22,100 +18,378 @@
+    logstring = 'n-a'
   end
-  # insert titlepage image
-  filecontents = File.read(epub_tmp_html).gsub(/(<section data-type="titlepage")/,"\\1 data-titlepage=\"yes\"")
-  File.open(epub_tmp_html, 'w') do |output| 
-    output.write filecontents
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.processxsl in a new method for this script; to return a result for json_logfile
+def addEBKhyperlinks(saxonpath, epub_tmp_html, strip_span_xsl, logkey='')
+  `java -jar "#{saxonpath}" -s:"#{epub_tmp_html}" -xsl:"#{strip_span_xsl}" -o:"#{epub_tmp_html}"`
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def readHtml(file, logkey='')
+	filecontents = File.read(file)
+	return filecontents
+rescue => logstring
+	return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def stripManualBreaks(filecontents, logkey='')
+  filecontents = filecontents.gsub(/(<p class="EBKLinkSourceLa">)(.*?)(<\/p>)(<p class="EBKLinkDestinationLb">)(.*?)(<\/p>)/,"\\1<a href=\"\\5\">\\2</a>\\3").gsub(/<br\/>/," ")
+  return filecontents
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def combineContiguousSpanUrls(filecontents, logkey='')
+  filecontents = filecontents.gsub(/(<span class="spanhyperlinkurl">)([^<|^>]*)(<\/span><span class="spanhyperlinkurl">)/,"\\1\\2")
+  return filecontents
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.runnode in a new method for this script; to return a result for json_logfile
+def localRunNode(jsfile, args, logkey='')
+	Bkmkr::Tools.runnode(jsfile, args)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.movesection in a new method for this script; to return a result for json_logfile
+def localMoveSection(inputfile, sectionparams, src, srcseq, dest, destseq, logkey='')
+	Bkmkr::Tools.movesection(inputfile, sectionparams, src, srcseq, dest, destseq)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.insertaddons in a new method for this script; to return a result for json_logfile
+def localInsertAddons(inputfile, sectionparams, addonparams, logkey='')
+	Bkmkr::Tools.insertaddons(inputfile, sectionparams, addonparams)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.compileJS in a new method for this script; to return a result for json_logfile
+def localCompileJS(file, logkey='')
+	Bkmkr::Tools.compileJS(file)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def biblioLookup(logkey='')
```

Comment:
change this to "databaseLookup", since we aren't actually connecting to Biblio itself